api/src/main/java/org/apache/iceberg/expressions/Literal.java

```diff
@@ -71,6 +73,14 @@ static Literal<BigDecimal> of(BigDecimal value) {
     return new Literals.DecimalLiteral(value);
   }
 
+  static Literal<Geometry> of(Geometry value) {
+    return new Literals.GeometryLiteral(value);
+  }
+
+  static Literal<Geography> of(Geography value) {
```

Comment:
I guess here, let's specifically not leak the Geography in the API, we should just have a wrapper class as well.